docs/en/query_language/agg_functions/reference.md

```diff
@@ -259,6 +259,60 @@ GROUP BY timeslot
 └─────────────────────┴──────────────────────────────────────────────┘
 ```
 
+## TSgroupSum(uid, timestamp, value) {#agg_function-tsgroupsum}
+TSgroupSum can aggregate different time series that sample timestamp not alignment.
+It will use linear interpolation between two sample timestamp and then sum time-series together.
+
+`uid` is the time series unique id, UInt64.
+`timestamp` is Int64 type in order to support millisecond or microsecond.
+`value` is the metric.
+
+Before use this function, timestamp should be in ascend order
+
+Example:
+```
+┌─uid─┬─timestamp─┬─value─┐
+│ 1   │     2     │   0.2 │
+│ 1   │     7     │   0.7 │
+│ 1   │    12     │   1.2 │
+│ 1   │    17     │   1.7 │
+│ 1   │    25     │   2.5 │
+│ 2   │     3     │   0.6 │
+│ 2   │     8     │   1.6 │
+│ 2   │    12     │   2.4 │
+│ 2   │    18     │   3.6 │
+│ 2   │    24     │   4.8 │
+└─────┴───────────┴───────┘
+```
+```
+CREATE TABLE time_series(
+    uid       UInt64,
+    timestamp Int64,
+    value     Float64
+) ENGINE = Memory;
+INSERT INTO time_series VALUES
+    (1,2,0.2),(1,7,0.7),(1,12,1.2),(1,17,1.7),(1,25,2.5),
+    (2,3,0.6),(2,8,1.6),(2,12,2.4),(2,18,3.6),(2,24,4.8);
+
+SELECT TSgroupSum(uid, timestamp, value)
+FROM (
+    SELECT * FROM time_series order by timestamp ASC
+);
+```
+And the result will be:
+```
+[(2,0.2),(3,0.9),(7,2.1),(8,2.4),(12,3.6),(17,5.1),(18,5.4),(24,7.2),(25,2.5)]
```

Comment:
why 0.2 at position 2 here? Looks like only value from timeseries uid=1 was taken.
 
Shouldn't series with uid=2 have  interpolated value 0.4 at position 2 ? And sum 0.2 + 0.4 = 0.6?

0.6 - (3-2) (1.6 - 0.6) / ( 8 - 3 )

Reply:
Same problem with last point.
Shouldn't it be  2.5+5 = 7.5 ?

```
┌──t─┬─ts1─┬─ts2─┬─sum─┐
│  2 │ 0.2 │ 0.4 │ 0.6 │
│  3 │ 0.3 │ 0.6 │ 0.9 │
│  7 │ 0.7 │ 1.4 │ 2.1 │
│  8 │ 0.8 │ 1.6 │ 2.4 │
│ 12 │ 1.2 │ 2.4 │ 3.6 │
│ 17 │ 1.7 │ 3.4 │ 5.1 │
│ 18 │ 1.8 │ 3.6 │ 5.4 │
│ 24 │ 2.4 │ 4.8 │ 7.2 │
│ 25 │ 2.5 │   5 │ 7.5 │
└────┴─────┴─────┴─────┘
```

Reply:
because at the beginning，only Time Series 1 starts，there is no Time Series 2, we only linear interpolation between two data point in one series

Reply:
I understand that, but isn't interpolation 'behind' those starting / ending point (using first two / last two known points) desirable? 

In many cases you will have start/end point just unaligned

Reply:
> I understand that, but isn't interpolation 'behind' those starting / ending point (using first two / last two known points) desirable?
> 
> In many cases you will have start/end point just unaligned
If Time series 2 really start at 3 and end at 24, we interpolation those starting / ending point (using first two / last two known points) , we will not get the real metrics at the time 2 and time 25, so we insist to interpolation between two data point in one series.

Yes, if our sensor sample metric with millisecond/microsecond timestamp every 60 seconds,  most every time series is unaligned. If we take granular intervals of 60 seconds to downsample each time series, that will be aligned, but the granularity is too large.

Reply:
How about using the initial value before the first timestamp & using the last value after the last one?
In that case, the sum of the first point will be 0.2 + 0.6 = 0.8, and the sum of the last point will be 2.5 + 4.8 = 7.3